packages/flame/lib/src/components/raster_sprite_component.dart

```diff
@@ -0,0 +1,48 @@
+import 'dart:async';
+
+import 'package:flame/components.dart';
+import 'package:meta/meta.dart';
+
+/// {@template raster_sprite_component}
+/// A [RasterSpriteComponent] is a [SpriteComponent] that
+/// will rasterize its sprite when loaded and will automatically
+/// manage the disposal of the rasterized image when removed.
+/// {@endtemplate}
+class RasterSpriteComponent extends SpriteComponent {
+  /// {@macro raster_sprite_component}
+  RasterSpriteComponent({
+    required this.baseSprite,
+    super.autoResize,
+    super.paint,
+    super.position,
+    super.size,
+    super.scale,
+    super.angle,
+    super.nativeAngle,
+    super.anchor,
+    super.children,
+    super.priority,
+    super.bleed,
+    super.key,
+  });
+
+  /// The base sprite to be rasterized.
+  final Sprite baseSprite;
+
+  @mustCallSuper
+  @override
+  FutureOr<void> onLoad() async {
+    await super.onLoad();
+
+    sprite = await baseSprite.rasterize();
+  }
+
+  @mustCallSuper
+  @override
+  void onRemove() {
+    super.onRemove();
+
+    sprite?.image.dispose();
+    sprite = null;
```

Comment:
There should be a warning that the component can't be reused after it was been removed, since `onLoad` won't be called again when the component is re-added. I wonder if this could be solved in some other way so that this wouldn't be an issue. :thinking:

Reply:
Hmmm, that is a very good point!

I thought on putting the raster image on cache, but I couldn't think on a good, generic enough solution for it, like what would we use as a key for the cache, how would we make sure that the raster image is properly disposed from cache?